dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -221,249 +253,282 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
                 }
             }
 
-            BackgroundTaskBuilder builder{ nullptr };
-
-            if (WI_IsFlagSet(registrationActivators, PushNotificationRegistrationActivators::PushTrigger) && PushNotificationHelpers::IsPackagedAppScenario())
+            winrt::guid registeredClsid{ GUID_NULL };
+            if (AppModel::Identity::IsPackagedProcess())
             {
-                GUID taskClsid = details.TaskClsid();
-                THROW_HR_IF(E_INVALIDARG, taskClsid == GUID_NULL);
+                // Returns ComActivator CLSID from registry. This CLSID provided in manifest is registered when a packaged app is installed
+                THROW_IF_FAILED(PushNotificationHelpers::GetComRegistrationFromRegistry(expectedPushServerArgs.data(), registeredClsid));
+            }
 
-                {
-                    auto lock = s_activatorInfoLock.lock_shared();
-                    THROW_HR_IF(E_INVALIDARG, s_pushTriggerRegistration);
-                }
+            auto scopeExitToCleanRegistrations{ wil::scope_exit(
+            [&](){
+                    auto lock{ m_lock.lock_exclusive() };
 
-                winrt::hstring taskClsidStr{ winrt::to_hstring(taskClsid) };
-                winrt::hstring backgroundTaskFullName{ backgroundTaskName + taskClsidStr };
+                    m_comActivatorRegistration.reset();
 
-                auto tasks{ BackgroundTaskRegistration::AllTasks() };
-                bool isTaskRegistered{ std::any_of(std::begin(tasks), std::end(tasks),
-                    [&](auto&& task)
+                    // Clean the task registration only if it was created during this call
+                    if (m_pushTriggerRegistration)
                     {
-                        auto name{ task.Value().Name() };
-
-                        if (std::wstring_view(name).substr(0, backgroundTaskName.size()) != backgroundTaskName)
-                        {
-                            return false;
-                        }
+                        m_pushTriggerRegistration.Unregister(true);
+                        m_pushTriggerRegistration = nullptr;
+                    }
+                }
+            ) };
 
-                        if (name == backgroundTaskFullName)
-                        {
-                            s_pushTriggerRegistration = task.Value();
-                            return true;
-                        }
+            /* Every packaged app is already getting the benefit of COM Registration to handle notifications.
+               However, we would benefit from power management improvements in Background Infrastructure.
+               So we are continuing to register a Push trigger post 19041 builds */
+               
+            if (PushNotificationHelpers::IsPackagedAppScenario())
+            {
+                {
+                    auto lock = m_lock.lock_shared();
+                    THROW_HR_IF(E_INVALIDARG, m_pushTriggerRegistration);
```

Comment:
THROW_HR_IF_MSG(E_INVALIDARG, m_pushTriggerRegistration, "The app has already Registered!");